velox/common/base/benchmarks/BitUtilBenchmark.cpp

```diff
@@ -141,6 +141,55 @@ BENCHMARK_RELATIVE(BM_scatterBits, n) {
   runScatterBits(n, false);
 }
 
+void BM_forEachBit(
+    uint32_t iterations,
+    size_t numEntries,
+    bool allSelected,
+    std::optional<size_t> bitPos = std::nullopt) {
+  folly::BenchmarkSuspender suspender;
+  uint32_t size = bits::nwords(numEntries);
+  std::vector<uint64_t> vec(size);
+  uint64_t* vecPtr = vec.data();
+  bits::fillBits(vecPtr, 0, numEntries, true);
+  if (!allSelected) {
+    bits::setBit(vecPtr, bitPos.value(), false);
+  }
+  suspender.dismiss();
+
+  for (uint32_t i = 0; i < iterations; ++i) {
+    bits::forEachBit(vecPtr, 0, numEntries, true, [&](auto row) {
+      return folly::doNotOptimizeAway(row * row);
+    });
+  }
+  suspender.rehire();
+}
+
+namespace {
+const static auto kNumRuns = 1000;
+}
+void BM_ForEachBitAllTrue(uint32_t iterations, size_t numEntries) {
+  for (int i = 0; i < kNumRuns; i++) {
```

Comment:
Why do you need both iterations and kNumRuns?

Reply:
iterations isnt big enough, we typically get values in micro or nano seconds, kNumRuns is just to amplify those runs.

Reply:
Why not just increase "iterations"?

Reply:
The absolute time per iteration does matter for sanity checking and for comparing to other benchmark results. For example, the cost per row here should affect the cost per row of almost every other function in the codebase.

For example: your current benchmark results are ~300us or ~3k-2k iters/sec. This is for 1000 x 10k blocks, so we are talking about 30-20B rows/sec, or 0.03ns/row. It's useful to have this as a datapoint when looking at the e.g. benchmarks of the simple UDF interface.

You can use the `_MULTI` family of functions to change how the iterations and time/iteration are displayed; IMO we should try to standardize on an iteration being a row in benchmark output.
https://github.com/facebook/folly/blob/main/folly/Benchmark.h#L392